File: II-OITF/p3/p3.py
import sys

for line in sys.stdin:
    x = line.strip().split()
    minutos = 0
    if x[5] == "fijo":
        if float(x[4]) >= float(x[0]):
            minutos = (float(x[4]) - float(x[0])) // float(x[2])
        else:
            minutos = 0
    else:
        if float(x[4]) >= float(x[1]):
            minutos = (float(x[4]) - float(x[1])) // float(x[3])
        else:
            minutos = 0
    if minutos == 0:
        print("Saldo insuficiente")
    else:
        print(int(minutos))

File: II-OITF/p3/p3.cpp
#include <iostream>
#include <sstream>
#include <string>

int main() {
    std::string line;
    while (std::getline(std::cin, line)) {
        std::istringstream iss(line);
        double coste_fijo, coste_movil, minuto_fijo, minuto_movil, saldo;
        std::string tipo;
        iss >> coste_fijo >> coste_movil >> minuto_fijo >> minuto_movil >> saldo >> tipo;

        int minutos = 0;
        if (tipo == "fijo") {
            if (saldo >= coste_fijo) {
                minutos = static_cast<int>((saldo - coste_fijo) / minuto_fijo);
            }
        } else {
            if (saldo >= coste_movil) {
                minutos = static_cast<int>((saldo - coste_movil) / minuto_movil);
            }
        }

        if (minutos == 0) {
            std::cout << "Saldo insuficiente" << std::endl;
        }
        else{
            std::cout << minutos << std::endl;
        }
    }
    return 0;
}